asdf/external_reference.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import absolute_import, division, unicode_literals, print_function
+
+from .asdftypes import AsdfType
+
+
+class ExternalArrayReference(AsdfType):
+    """
+    Store a reference to an array in an external File.
+
+    Parameters
+    ----------
+
+    filepath: `str`
+        The path to the path to be referenced. Can be relative to the file
```

Comment:
Maybe this should actually be a URI since this might be useful for referencing files that don't necessarily live on the host file system.